packages/nextjs/app/page.tsx

```diff
@@ -1,67 +1,59 @@
 "use client";
 
-import Link from "next/link";
+import { useRouter } from "next/navigation";
 import type { NextPage } from "next";
-import { useAccount } from "wagmi";
-import { BugAntIcon, MagnifyingGlassIcon } from "@heroicons/react/24/outline";
-import { Address } from "~~/components/scaffold-eth";
+import { PlusCircleIcon } from "@heroicons/react/24/outline";
+import { AuditList } from "~~/components/AuditList";
+import { useHydrateDraftAudits } from "~~/services/store";
 
 const Home: NextPage = () => {
-  const { address: connectedAddress } = useAccount();
+  const router = useRouter();
+  const { draftAudits, addDraftAudit, hydrated } = useHydrateDraftAudits();
+
+  function createNewAudit() {
+    const defaultTitle = "Untitled_Audit";
+
+    const numbers = draftAudits.map(audit => {
+      const match = audit.id.match(/audit_(\d+)/);
+      return match ? parseInt(match[1], 10) : -1;
+    });
+
+    const nextId = numbers.length ? Math.max(...numbers) + 1 : 1;
+
+    const id = `audit_${nextId.toString()}`;
```

Comment:
I would suggest to use [randomUUID](https://developer.mozilla.org/en-US/docs/Web/API/Crypto/randomUUID) here, eg
```suggestion
    const id = crypto.randomUUID();
```
then you don't need all logic of determining next id